db/dcrpg/internal/vinoutstmts.go

```diff
@@ -58,7 +61,16 @@ const (
 	SelectFundingOutpointByVinID = `SELECT prev_tx_hash, prev_tx_index, prev_tx_tree FROM vins WHERE id=$1;`
 	SelectFundingTxByVinID       = `SELECT prev_tx_hash FROM vins WHERE id=$1;`
 	SelectSpendingTxByVinID      = `SELECT tx_hash, tx_index, tx_tree FROM vins WHERE id=$1;`
-	SelectAllVinInfoByID         = `SELECT * FROM vins WHERE id=$1;`
+	SelectAllVinInfoByID         = `SELECT id, tx_hash, tx_index, tx_tree, is_valid, block_time,
+		prev_tx_hash, prev_tx_index, prev_tx_tree, value_in FROM vins WHERE id = $1;`
+	SetIsValidByTxHash = `UPDATE vins SET is_valid = $1 WHERE tx_hash = $2;`
+
+	SetVinsTableCoinSupplyUpgrade = `UPDATE vins SET is_valid = $1, block_time = $2, value_in = $3
+		WHERE tx_hash = $4 and tx_index = $5 and tx_tree = $6;`
```

Comment:
same as above.  This tx hash could be in an orphaned block.

tx_index just says which transaction number within the containing block it is.

Reply:
`tx_hash, tx_index, and tx_tree` are unique for every vin entry.

Reply:
Ok, same as other comment.  I'm there transactions table is not unique for the reason I mentioned, but that doesn't apply.  My mistake